src/LightGBM.jl

```diff
@@ -1,13 +1,38 @@
 module LightGBM
 
 using Dates
-
 import Base
 import Libdl
 import StatsBase
+import Libdl
+
+struct LibraryNotFoundError <: Exception
+    msg::String
+end
+
+
+function find_library(library_name::String, custom_paths::Vector{String})
+
+    # Search system filedirs first, returns empty string if not found
+    output = Libdl.find_library(library_name)
+
+    if output == ""
+        # try specified paths
+        @info("$(library_name) not found in system dirs, trying fallback")
+        output = Libdl.find_library(library_name, custom_paths)
+    else
+        @info("$(library_name) found in system dirs!")
+    end
+
+    if output == ""
+        throw(LibraryNotFoundError("$(library_name) not found. Please ensure this library is either in system dirs or the dedicated paths: $(custom_paths)"))
```

Comment:
we might have to go back against this later if it causes problems with compiling system images, but i dont know enough at this stage